pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,296 @@
+package workflows_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+)
+
+// Note that the set of tests in this file cover the conversion from existing YAM -> code
+// along with testing the structure of what is generated from the builders.
+// This implicitly tests the code generators functionally, as the generated code is used in the tests.
+
+type Config struct {
+	Workflow    workflows.NewWorkflowParams
+	Streams     *streams.TriggerConfig
+	Ocr         *ocr3.ConsensusConfig
+	ChainWriter *chainwriter.TargetConfig
+	TargetChain string
+}
+
+func NewWorkflowSpec(rawConfig []byte) (workflows.WorkflowSpec, error) {
+	conf := Config{}
+	if err := yaml.Unmarshal(rawConfig, &conf); err != nil {
+		return workflows.WorkflowSpec{}, err
+	}
```

Comment:
I'm not familiar with this yaml lib, but it looks like we could wrap it with generic helpers. Also, changing the signature to return `*Workflow` or an interface would pay off here too:
```suggestion
	conf, err := UnmarshalYaml[Config](rawConfig)
	if err != nil {
		return nil, err
	}
```
```go
func UnmarshalYaml[T any](raw []byte) (T, error) {
	var v T
	err := yaml.Unmarshal(raw, &v)
	return v, err 
}
```